modules/gdscript/gdscript_parser.cpp

```diff
@@ -2183,6 +2183,9 @@ GDScriptParser::IfNode *GDScriptParser::parse_if(const String &p_token) {
 		push_error(vformat(R"(Expected conditional expression after "%s".)", p_token));
 	}
 
+	if (check(GDScriptTokenizer::Token::PERIOD_PERIOD)) {
+		push_error(R"(Expected statement, found ".." instead.)");
+	}
 	consume(GDScriptTokenizer::Token::COLON, vformat(R"(Expected ":" after "%s" condition.)", p_token));
```

Comment:
```suggestion
	if (!check(GDScriptTokenizer::Token::PERIOD_PERIOD)) {
        consume(GDScriptTokenizer::Token::COLON, vformat(R"(Expected ":" after "%s" condition.)", p_token));
	}
```
This will make it so the error message about `":"` does not get pushed and have `Expected statement, found ".." instead.` as the last error message in the list. After that squash the commits.

Reply:
i dont see where the squash is at. i do think there is a squash and merge option when the time comes for it?